src/protocol/boolean/random_bits.rs

```diff
@@ -0,0 +1,170 @@
+use crate::error::BoxError;
+use crate::ff::{Field, Int};
+use crate::protocol::boolean::BitOpStep;
+use crate::protocol::modulus_conversion::convert_shares::{ConvertShares, XorShares};
+use crate::protocol::{context::ProtocolContext, RecordId};
+use crate::secret_sharing::Replicated;
+use futures::future::try_join_all;
+
+/// This protocol generates a sequence of uniformly random sharing of bits in `F_p`.
+/// Adding these 3-way secret-sharing will yield the secret `r_i ∈ {0,1}`.
+pub struct RandomBits {}
+
+impl RandomBits {
+    #[allow(dead_code)]
+    pub async fn execute<F: Field>(
+        ctx: ProtocolContext<'_, Replicated<F>, F>,
+        record_id: RecordId,
+    ) -> Result<(Vec<Replicated<F>>, Replicated<F>), BoxError> {
+        // We assume the bit length we operate in would not exceed 255
+        #[allow(clippy::cast_possible_truncation)]
+        let l = F::Integer::BITS as u8;
+
+        // Generate a pair of random numbers. We'll use these numbers as
+        // the source of `l`-bit long uniformly random sequence of bits.
+        let r_bits: (F, F) = ctx
```

Comment:
```suggestion
        let (rbits_left, rbits_right): (F, F) = ctx
```

Please, if you are doing `x.0` or `x.1`, try to think of that as a problem, not a feature.

Reply:
Oh, I wasn't aware. Thanks!